go/worker/event/server.go

```diff
@@ -299,28 +300,98 @@ func Main() (err error) {
 		return fmt.Errorf("unknown Server_mode %s", common.Conf.Server_mode)
 	}
 	if err != nil {
-		os.Remove(pidPath)
 		return err
 	}
 
-	// clean up if signal hits us (e.g., from ctrl-C)
-	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt, syscall.SIGTERM)
-	signal.Notify(c, os.Interrupt, syscall.SIGINT)
-	go func() {
-		<-c
-		slog.Info("Received kill signal, cleaning up.")
-		shutdown(pidPath, s)
+	// sock file is made in worker directory
+	sockPath := filepath.Join(common.Conf.Worker_dir, "ol.sock")
+	// remove sock file on exit
+	defer func() { 
+		slog.Info(fmt.Sprintf("Remove sockPath %s.", sockPath))
+		if err := os.Remove(sockPath); err != nil {
+			slog.Error(fmt.Sprintf("Remove sockPath error: %s", err))
+		}
 	}()
 
+	// worker access sock file
+	ln, errUDS := net.Listen("unix", sockPath)
+	if errUDS != nil {
+		return fmt.Errorf("failed to listen on UNIX domain socket %s: %w", sockPath, errUDS)
+	}
+	if err := os.Chmod(sockPath, 0o600); err != nil {
+		_ = ln.Close()
+		return fmt.Errorf("chmod UNIX domain socket sock file %s: %w", sockPath, err)
+	}
+	
+	udsServer := &http.Server{
+		Handler:           udsMux,
+		ReadHeaderTimeout: 5 * time.Second,
+	}
+	
 	port := fmt.Sprintf("%s:%s", common.Conf.Worker_url, common.Conf.Worker_port)
-	err = http.ListenAndServe(port, nil)
+	portServer := &http.Server{
+		Addr:    port,
+		Handler: portMux,
+	}
+
+	// list of servers so all shutdown logic can be in one place
+	servers := map[string] * http.Server{
+		"uds": udsServer,
+		"tcp": portServer,
+	}
+
+	// error channel to handle server errors
+	errorChannel := make(chan error, len(servers))
+	// error channel to handle signals (e.g., from ctrl-C)
+	signalChannel := make(chan os.Signal, 1)
+	signal.Notify(signalChannel, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
 
-	// if ListenAndServer returned, there must have been some issue
-	// (probably a port collision)
+	// start serving on the UNIX domain socket server
+	go func() {
+		slog.Info("worker listening on UNIX domain socket", "socket", sockPath)
+		err := udsServer.Serve(ln)
+		if err != nil && err != http.ErrServerClosed {
```

Comment:
When we check if != nil like this, it feels to the reader that nil is a valid return value.  But the documentation says it is not.  We should quote the documentation here, and either panic if we get nil (since it violates our assumptions), or send an error saying something like "Server unexpectedly returned nil".